bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookieWriteLedgerTest.java

```diff
@@ -392,6 +477,79 @@ public void testLedgerCreateAdvWithLedgerIdInLoop() throws Exception {
         }
     }
 
+    /**
+     * In a loop create/write/read/delete the ledger with ledgerId through the
+     * functionality of Advanced Ledger which accepts ledgerId as input.
+     * In this testcase (other testcases don't cover these conditions, hence new
+     * testcase is added), we create entries which are greater than
+     * SKIP_LIST_MAX_ALLOC_ENTRY size and tried to addEntries so that the total
+     * length of data written in this testcase is much greater than
+     * SKIP_LIST_SIZE_LIMIT, so that entries will be flushed from EntryMemTable
+     * to persistent storage
+     *
+     * @throws Exception
+     */
+    @Test(timeout = 180000)
+    public void testLedgerCreateAdvWithLedgerIdInLoop2() throws Exception {
+
+        assertTrue("Here we are expecting Bookies are configured to use SortedLedgerStorage",
+                baseConf.getSortedLedgerStorageEnabled());
+
+        long ledgerId;
+        int ledgerCount = 10;
+
+        List<List<byte[]>> entryList = new ArrayList<List<byte[]>>();
+        LedgerHandle[] lhArray = new LedgerHandle[ledgerCount];
+        long skipListSizeLimit = baseConf.getSkipListSizeLimit();
```

Comment:
this is not a good unit test. can you set the skip list size limit to a smaller value? you shouldn't use the default value of testing purpose.

Reply:
im not quite sure if it is easy/worthy enough to change the conf value just for this testcase. The issue I see here is BookKeeper cluster is started as part of @Before method and the conf values should be set in this method. I dont see a way to have customized conf value just for this testcase.

Reply:
you have a few options here:

1) what is the side effects of setting smaller skip list size limit? if there is not side effects, you can set the setting in the constructor, and the same setting will be applied to all the test cases. Ideally all the other tests should run okay with any skip list size limit.

2) you can stop the cluster, update the configuration and start the cluster in this test case.

3) you can move this test case to a separate test suite, which is configured with smaller skip list limit size.

thoughts?

Reply:
done, set it in the constructor.